### PR TITLE
feat: added required urls for feature lists

### PR DESCRIPTION
**Pull request by sowmyadev-kota**
#### Description of changes:

This will add the content URLS for `featurelist` components

We can successfully verify the links since they are added as relative thanks to @reesscot 

Staging Site: https://featurte-list-urls.d1ywzrxfkb9wgg.amplifyapp.com/

Some test cases 

- React native is using the same as the other JS pages (enable-sign-up). Only swift, android, and flutter use the enable-sign-in page

- Cloud-connected UI components of Storage available only for  `React/Javascript`.

image

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/constants/feature-lists-data.ts [hunk 1]
```diff
@@ -1,35 +1,209 @@
...
+            link: 'build-a-backend/graphqlapi/subscribe-data/'
+          },
+          {
+            content:
+              'Implement fine-grained authorization rules and policies to securely access data. Control auth at the API level or on individual fields. Manage access with AWS IAM policies or your own auth provider.',
+            linkText: 'Granular Authorization',
+            link: 'build-a-backend/graphqlapi/customize-authorization-rules/'
+          }
+        ],
+        heading: 'Data'
```
Comment (renebrandel) [inline src/constants/feature-lists-data.ts:50]: We need to remove the label "Data" for current gen content and only use it for Gen 2. Use "GraphQL API" instead.

Comment (sowmyadev-kota) [reply src/constants/feature-lists-data.ts:50]: Got it, thank you

Comment (sowmyadev-kota) [reply src/constants/feature-lists-data.ts]: Updated @renebrandel

**Review by renebrandel — APPROVED** [review]

**Review by renebrandel — DISMISSED** [review]

File: src/components/FeatureLists/PlatformFeatureList.tsx [hunk 2]
```diff
@@ -25,8 +27,14 @@ const PlatformFeatureList: React.FC<PlatformFeatureListProps> = ({
...
-                href={categoryItem.link}
-                isExternal={true}
+                href={
+                  platform === 'javascript' &&
+                  pathname === '/' &&
+                  !categoryItem.link.startsWith('https://ui.docs.amplify.aws/')
+                    ? `${platform}/${categoryItem.link}`
+                    : categoryItem.link
+                }
+                isExternal={categoryItem.isExternal === true ? true : false}
```
Comment (hbuchel) [inline src/components/FeatureLists/PlatformFeatureList.tsx:37]:
```suggestion
                isExternal={categoryItem.isExternal}
```

Comment (sowmyadev-kota) [reply src/components/FeatureLists/PlatformFeatureList.tsx:37]: Nice

Comment (sowmyadev-kota) [reply src/components/FeatureLists/PlatformFeatureList.tsx]: Done @hbuchel !

File: src/components/FeatureLists/PlatformFeatureList.tsx [hunk 3]
```diff
@@ -25,8 +27,14 @@ const PlatformFeatureList: React.FC<PlatformFeatureListProps> = ({
...
                 linkText={categoryItem.linkText}
-                href={categoryItem.link}
-                isExternal={true}
+                href={
+                  platform === 'javascript' &&
+                  pathname === '/' &&
+                  !categoryItem.link.startsWith('https://ui.docs.amplify.aws/')
+                    ? `${platform}/${categoryItem.link}`
+                    : categoryItem.link
+                }
```
Comment (hbuchel) [inline src/components/FeatureLists/PlatformFeatureList.tsx:36]:
```suggestion
                href={
                  platform === 'react' &&
                  pathname === '/' &&
                  !categoryItem.isExternal
                    ? `${platform}/${categoryItem.link}`
                    : categoryItem.link
                }
```

Your PR isn't up to date with next-release/main, but react is the default platform now. We should find a more elegant way to do this so we have less places to hard code the default platform, but I tested locally against latest and this worked for me.

**Review by hbuchel — CHANGES_REQUESTED** [review]
Thanks @sowmyadev-kota  Left a change inline, feel free to ping me after it's updated.

**Comment by sowmyadev-kota** [comment]
> Thanks @sowmyadev-kota Left a change inline, feel free to ping me after it's updated.

Used  `DEFAULT_PLATFORM` so we don't have to hard code here :) thanks @hbuchel

File: src/components/FeatureLists/PlatformFeatureList.tsx (hunk 3 above)
Comment (sowmyadev-kota) [reply src/components/FeatureLists/PlatformFeatureList.tsx]: Used DEFAULT_PLATFORM so we don't have to hard code here :) thanks

Comment (hbuchel) [reply src/components/FeatureLists/PlatformFeatureList.tsx]: NICE 🔥 thank you!

**Review by hbuchel — APPROVED** [review]
Tested locally and with next-release merged in to test react as default platform and LGTM!

**Review by renebrandel — APPROVED** [review]

**Pull request merged**